app/src/main/java/com/example/polina/socialnetwork/ProfileActivity.java

```diff
@@ -153,19 +181,30 @@ public void addProfileInfo(JSONObject o) {
 
     @UiThread
     public void updateAdapter(ArrayList<Post> posts) {
-        postList.setAdapter(adapter);
-        adapter.clear();
         adapter.addAll(posts);
         adapter.notifyDataSetChanged();
+        postsToLoad.addAll(posts);
+        count = 0;
     }
 
     private void loadProfileFromMemory(SharedPreferences sharedPreferences) {
-        name.setText(sharedPreferences.getString(ServerAPI.NAME, ""));
-        int y = Utils.calculateAmountYears(sharedPreferences.getString(ServerAPI.BIRTHDAY, ""));
+        name.setText(sharedPreferences.getString(Utils.NAME, ""));
+        int y = Utils.calculateAmountYears(sharedPreferences.getString(Utils.BIRTHDAY, ""));
         String years = getResources().getQuantityString(R.plurals.years, y, y);
         birthday.setText(years);
-        profileURL = sharedPreferences.getString(ServerAPI.PROF_URL, "");
-        mImageLoader.get(profileURL, ImageLoader.getImageListener(image, 0, 0));
+        profileURL = sharedPreferences.getString(Utils.PROF_URL, "");
+        mImageLoader.get(profileURL, ImageLoader.getImageListener(image, R.drawable.load, R.drawable.error));
```

Comment:
for better performance please use NetworkImageView class(should be changed in layout file also). 
For ex.: image.setImageUrl(profileURL, mImageLoader);